### PR TITLE
[Mercurial] profits beneficiary

### PR DESCRIPTION
**Pull request by cnek**
much similar to #36 
except:
- try to keep `MercurialVaultDepository` store only immutable property, `profitsBeneficiaryKey` should only be accessed through reading on chain account on `MercurialVaultDepositoryAccount`

program PR : https://github.com/UXDProtocol/uxd-program/pull/222

### CODE REVIEW

File: src/idl.ts [hunk 1]
```diff
@@ -1332,89 +1332,83 @@ export type Uxd = {
...
+            '#6',
             'Token account holding the LP tokens minted by depositing collateral on mercurial vault'
           ];
         },
         {
           name: 'mercurialVault';
           isMut: true;
           isSigner: false;
-          docs: ['#8'];
+          docs: ['#7'];
```
Comment (crypto-vincent) [inline src/idl.ts]: I didnt know the docs were included on our version of anchor, I thought that didnt work, but if it does work its nice!

**Review by crypto-vincent — APPROVED** [review]

File: src/idl.ts (hunk 1 above)
Comment (cnek) [reply src/idl.ts]: no it didn't work, would hv deserialisating error with program build with older version anchor like us, thx for reminding

**Comment by crypto-vincent** [comment]
qq: should we merge this @cnek ?

File: src/mercurial/utils.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import { ParsedClockState, VaultState } from './types';
+
+export const LOCKED_PROFIT_DEGRADATION_DENOMINATOR = new BN(1_000_000_000_000);
+
+export const getOnchainTime = async (connection: Connection) => {
+  const parsedClock = await connection.getParsedAccountInfo(
+    SYSVAR_CLOCK_PUBKEY
+  );
+
+  const parsedClockAccount = (parsedClock.value!.data as ParsedAccountData)
```
Comment (crypto-vincent) [inline src/mercurial/utils.ts:17]: github gives a warning to this line

File: src/mercurial/depository.ts [hunk 3]
```diff
@@ -138,8 +137,50 @@ export class MercurialVaultDepository {
       vaultPda,
       mercurialVaultLpMint,
       depositoryLpTokenVault,
-      mercurialVaultCollateralTokenSafe
+      mercurialVaultCollateralTokenSafe,
+      mercurialVaultProgram
+    );
+  }
+
+  public async calculateProfitsValue(
```
Comment (crypto-vincent) [inline src/mercurial/depository.ts:145]: to speed up the merge of this branch, we could put this profit calculation function into another future incremental PR (we wouldnt need types/utils new files for this IDL change)

Comment (cnek) [reply src/mercurial/depository.ts:145]: it's been using in the test on uxd-program already for mercurial profit collection, let's include that in this PR

File: src/mercurial/utils.ts (hunk 2 above)
Comment (cnek) [reply src/mercurial/utils.ts:17]: disable lint on that line

**Review by crypto-vincent — APPROVED** [review]
LGTM!

**Pull request merged**